### PR TITLE
banking fixed

### PR DESCRIPTION
**Pull request by madwizzard**
Screenshot 2024-07-10 083058
Screenshot 2024-07-10 083156

### CODE REVIEW

**Review by Juanro-Minne — DISMISSED** [review]

File: src/app/components/hris/employees/employee-profile/accordions/accordion-banking/accordion-banking.component.ts [hunk 1]
```diff
@@ -164,53 +164,58 @@ export class AccordionBankingComponent {
...
+        branch: `${employeeBankingFormValue.branch}`,
+        accountNo: `${employeeBankingFormValue.accountNo}`,
+        accountType: employeeBankingFormValue.accountType,
+        status: 1,
+        declineReason: this.bankingReason,
+        file: employeeBankingFormValue.file
+      }
+      if (this.hasBankingData) {
+        this.employeeBankingService.updatePending(this.employeeBankingDto).subscribe({
+          next: () => {
```
Comment (ZuenTheMan) [inline src/app/components/hris/employees/employee-profile/accordions/accordion-banking/accordion-banking.component.ts:184]: Can the code in the `next()` of adding and updating banking information not be placed in a function to call? Seems to me similar set of code.

**Review by ReinoRabbit — APPROVED** [review]

File: src/app/components/hris/employees/employee-profile/accordions/accordion-banking/accordion-banking.component.ts [hunk 2]
```diff
@@ -164,53 +164,58 @@ export class AccordionBankingComponent {
...
+            this.snackBarService.showSnackbar("Updated", "snack-success");
+            this.getEmployeeBankingData();
+            this.checkBankingInformationProgress();
+            this.totalBankingProgress();
+            this.employeeBankingStarterkitService.getAllBankingAndStarterkits();
```
Comment (ZuenTheMan) [inline src/app/components/hris/employees/employee-profile/accordions/accordion-banking/accordion-banking.component.ts:189]: Should this not be called in the `addBankingDetails ...next()` callback as well?

Comment (madwizzard) [reply src/app/components/hris/employees/employee-profile/accordions/accordion-banking/accordion-banking.component.ts:189]: I am not sure what you are talking about.

File: src/app/components/hris/employees/employee-profile/accordions/accordion-banking/accordion-banking.component.ts [hunk 3]
```diff
@@ -164,56 +164,58 @@ export class AccordionBankingComponent {
...
+      this.snackBarService.showSnackbar("Add a Proof of account", "snack-error")
     }
   }
 
+  addOrUpdateBanking(message: string) {
+    this.snackBarService.showSnackbar(message, "snack-success");
+    this.getEmployeeBankingData();
+    this.checkBankingInformationProgress();
+    this.totalBankingProgress();
+    this.employeeBankingStarterkitService.getAllBankingAndStarterkits();
```
Comment (EduanO) [inline src/app/components/hris/employees/employee-profile/accordions/accordion-banking/accordion-banking.component.ts:213]: For updatePending
```suggestion
    this.employeeBankingStarterkitService.getAllBankingAndStarterkits();
```
For addBankingDetails
```suggestion
    this.employeeBankingStarterkitService.incrementPendingCount();
```

**Review by EduanO — CHANGES_REQUESTED** [review]

**Review by Juanro-Minne — APPROVED** [review]

**Review by EduanO — APPROVED** [review]

**Review by ReinoRabbit — APPROVED** [review]

**Pull request merged**